railties/lib/rails/command/base.rb

```diff
@@ -177,9 +177,12 @@ def namespaced_commands
       end
 
       no_commands do
-        delegate :executable, to: :class
         attr_reader :current_subcommand
 
+        def executable(subcommand = nil)
+          self.class.executable(subcommand)
+        end
+
```

Comment:
Some of these call sites do not seem relevant to improving request / response performance.  For example, this call site is only relevant to CLI commands.

Reply:
Right. Only the first and second commits on this PR are relevant to request / response performance, and the rest are about CLI commands and AR querying respectively. With these four commits, I could cover all delegations that were detected on a basic scaffold app. I can simply drop the third commit from this PR though if that's easier for you to review.